aptos-move/framework/aptos-framework/sources/account/account.move

```diff
@@ -424,6 +443,185 @@ module aptos_framework::account {
+        let backup_key_as_single_key = single_key::from_ed25519_public_key_unvalidated(&backup_key_ed25519);
+
+        // Construct the multi key public key which should be the current public key of the account
+        let new_public_key_bytes = vector[];
+        new_public_key_bytes.push_back(0x02); // Number of keys in the multi key
+        new_public_key_bytes.append(keyless_public_key);  // Add the current key which is the keyless public key
+        new_public_key_bytes.append(single_key::unvalidated_public_key_to_bytes(&backup_key_as_single_key));
+        new_public_key_bytes.push_back(0x01); // Signatures required
+
+        let new_auth_key = get_authentication_key_from_scheme_and_public_key_bytes(MULTI_KEY_SCHEME, new_public_key_bytes);
+
+        // Rotate the authentication key to the new multi key public key
+        rotate_authentication_key_call(account, new_auth_key);
+
+        event::emit(KeyRotationToMultiPublicKey {
+            account: addr,
+            verified_public_key_bit_map: vector[0xC0, 0x00, 0x00, 0x00],
+            multi_public_key_scheme: MULTI_KEY_SCHEME,
+            multi_public_key: new_public_key_bytes,
+            authentication_key: new_auth_key,
+        });
+    }
+
+    /// Replaces the ED25519 backup key on a keyless account that already has a backup key configured.
+    /// This function takes a keyless account with an existing backup key and replaces it with a new ED25519 backup key,
+    /// maintaining the multi-key configuration that requires 1 signature from either key to authenticate.
+    ///
+    /// # Arguments
+    /// * `account` - The signer representing the keyless account
+    /// * `keyless_public_key` - The current keyless public key of the account
+    /// * `cur_backup_key` - The current ED25519 backup public key to be replaced
+    /// * `new_backup_key` - The new ED25519 public key to replace the current backup key
+    /// * `new_backup_key_proof` - A signature from the new backup key proving ownership
+    ///
+    /// # Aborts
+    /// * If the main public key is not a keyless public key
+    /// * If the constructed multi-key (keyless + current backup) does not match the account's current authentication key
+    /// * If the keyless public key is not the original public key of the account
+    /// * If the new backup key proof signature is invalid
+    ///
+    /// # Events
+    /// * Emits a `KeyRotationToMultiPublicKey` event with the updated multi-key configuration
+    entry fun replace_ed25519_backup_key_on_keyless_account(account: &signer, keyless_public_key: vector<u8>, cur_backup_key: vector<u8>, new_backup_key: vector<u8>, new_backup_key_proof: vector<u8>) acquires Account {
+        // Check that the main public key is a keyless public key by checking the scheme
+        // encoded in the first byte of the single key public key
+        let public_key_type = keyless_public_key[0];
```

Comment:
This should be refactored into a common util function since it's called in multiple places. Will help reduce or eliminate the comment as well with a self-documenting name